### PR TITLE
Elapsed time stopwatch at Presenter mode

### PR DESCRIPTION
**Pull request by otorrillas**
Targeting #292 issue:

- Implemented a timer mode with `Start`/`Stop`/`Reset` buttons control, counting the elapsed time through the presentation.
- Added controls (`Alt/Option + T`) to toggle between normal presentation mode (current time) and timer presentation mode.
- Re-structured the time section at the `Presenter` component:
   - Created a `Time` component (`time.js`) to manage the toggle between the clock and the timer.
   - Moved the clock logic from `presenter.js` to a new `Clock` component (`components/clock.js`).
   - Created a `Timer` component (`timer.js) to handle the stopwatch logic and its control buttons. 
 

See screenshots below: 

screen shot 2017-04-05 at 14 44 27
screen shot 2017-04-05 at 15 03 11
screen shot 2017-04-05 at 15 03 21

### CODE REVIEW

File: src/components/timer.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    super();
+    this.state = {
+      elapsedTime: 0,
+      paused: true
+    };
+    this.handleStartTimer = this.handleStartTimer.bind(this);
+    this.handleStopTimer = this.handleStopTimer.bind(this);
+    this.handleResetTimer = this.handleResetTimer.bind(this);
+  }
+  componentDidMount() {
```
Comment (shaunwallace) [inline src/components/timer.js]: remove this empty block :-)

File: src/components/clock.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import React, {
+    Component
```
Comment (stefvhuynh) [inline src/components/clock.js]: Can we put this on one line so that's uniform with the rest of the code?

File: src/components/clock.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  hours %= 12;
+  hours = hours ? hours : 12;
+  minutes = minutes < 10 ? `0${minutes}` : minutes;
+  seconds = seconds < 10 ? `0${seconds}` : seconds;
+  const strTime = `${hours} : ${minutes} : ${seconds} ${ampm}`;
+  return strTime;
+};
+
+export default class Clock extends Component {
+
```
Comment (stefvhuynh) [inline src/components/clock.js]: Unnecessary space

File: src/components/time.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import React, {
+    Component,
+    PropTypes
```
Comment (stefvhuynh) [inline src/components/time.js]: Same thing here.

File: src/components/timer.js [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+    this.setState({
+      elapsedTime: 0
+    });
+  }
+  _renderResetButton() {
+    return (
+      <TSingleButton
+        onClick={this.handleResetTimer}
+      >
+      Reset
```
Comment (stefvhuynh) [inline src/components/timer.js]: indent

File: src/components/timer.js [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+    );
+  }
+
+  _renderStopButton() {
+    return (
+      <TSingleButton
+        onClick={this.handleStopTimer}
+        stop
+      >
+      Stop
```
Comment (stefvhuynh) [inline src/components/timer.js]: indent

**Comment by stefvhuynh** [comment]
lgtm. just a few nitpicky style things.

**Comment by shaunwallace** [comment]
@stefvhuynh @kenwheeler have you guys thought about introducing lint-staged and prettier to enforce consistent code styling?

**Comment by otorrillas** [comment]
@stefvhuynh fixed the code-styling comments. Hope it is good now.

**Comment by stefvhuynh** [comment]
:+1:

**Pull request merged**